### ISSUE TITLE
I can ping and apt-get using ipv6, but not ipv4

### ISSUE THREAD

**Issue by travelerjjm** [microsoft/WSL#2423]
[strace.txt](https://github.com/Microsoft/BashOnWindows/files/1223578/strace.txt)
When I do an `apt-get`, it goes through all the v4 addresses and fails on each, tries a v6 address and the `apt-get` succeeds. So I tried pinging google.com: no responses. I tried curl: no response.  I did an strace of ping. It is attached.  The summary is that the call to `recvmsg` fails with `EAGAIN` (-1) Resource temporarily unavailable. It is using the correct IP address, and dig works correctly. I tried googling and didn't see this issue. My windows is 10.0.15063 and I just updated to the current Ubuntu and did `apt-get upgrade`.

The only unusual trait of my system is that I have five ethernet adapters that show up. eth4 is the correct one (others talk to vmware VMs) and indeed the default route for v4 is on eth4, and I can ping that router. (Yes, `route -6` shows the default v6 through eth4, also.)

Thanks.

**Comment by sunilmut** [comment]
@travelerjjm - Thanks for your post. I really don't see anything else wrong with your strace, other than the 'recvmsg' calls failing (as you also point out). Your DNS resolution seems to be working fine. Can you share a strace of `ping6 google.com`?

Also, just to confirm, `route` is correctly showing `eth4` as the default route for IPv4?

**Comment by travelerjjm** [comment]
Thanks.
I made the `ping6` trace and went through it and wondered just what the issue might be with the ping. 
[ping6-trace.txt](https://github.com/Microsoft/BashOnWindows/files/1229351/ping6-trace.txt)
Here is the route output, too.
[route.txt](https://github.com/Microsoft/BashOnWindows/files/1229353/route.txt)

So I started Wireshark and looked at traffic along with firewall logs. I improved  the `curl` and `apt-get` issues by explicitly allowing traffic from TCP port 0, which is what the firewall log thinks BoW was using even though that is not what was on the wire! So I explicitly enabled all ICMP, too and that enabled ping.

BUT it seems v6 is now the default for `curl` and `apt-get`. I tried `curl` to a site without v6 support and it failed. Here is the strace:
[strace-curl.txt](https://github.com/Microsoft/BashOnWindows/files/1229346/strace-curl.txt)
Here is the firewall log entry (ZoneAlarm)
`FWOUT,2017/08/16,13:40:12 -6:00 GMT,10.4.0.100:35080,199.85.219.100:80,TCP (flags:S)`
I cannot see what could be bad there. So I enabled all TCP out, and now it seems to work.

I am left with questions: why did enabling TCP port 0 allow some sites to work by forcing them to v6? Why did explicitly enabling TCP and UDP allow TCP over v4 to work (when it already did over v6)? There is clearly some odd interaction between my firewall and BoW. I am not sure why. Maybe this will help others. I can live with allowing TCP out of my Windows 10, PC, LOL.

Thanks. If you hadn't said the initial strace looked OK, I might not have been able to fix it. I know I'd tried with the firewall off before, but that was likely before the upgrade to the new BoW.
